File: dx-web/src/consts/game-degree.ts
export const GAME_DEGREES = {
  BEGINNER: "beginner",
  INTERMEDIATE: "intermediate",
  ADVANCED: "advanced",
} as const;

export type GameDegree = (typeof GAME_DEGREES)[keyof typeof GAME_DEGREES];

export const GAME_DEGREE_LABELS: Record<GameDegree, string> = {
  beginner: "初级",
  intermediate: "中级",
  advanced: "高级",
};

export const DEGREE_CONTENT_TYPES: Record<GameDegree, string[] | null> = {
  beginner: null,
  intermediate: ["block", "phrase", "sentence"],
  advanced: ["sentence"],
};

File: dx-api/app/consts/game_degree.go
package consts

// Game degree values.
const (
	GameDegreeBeginner     = "beginner"
	GameDegreeIntermediate = "intermediate"
	GameDegreeAdvanced     = "advanced"
)

// GameDegreeLabels maps each degree to its Chinese label.
var GameDegreeLabels = map[string]string{
	GameDegreeBeginner:     "初级",
	GameDegreeIntermediate: "中级",
	GameDegreeAdvanced:     "高级",
}

// AllGameDegrees contains all game degree values.
var AllGameDegrees = []string{
	GameDegreeBeginner,
	GameDegreeIntermediate,
	GameDegreeAdvanced,
}

// DegreeContentTypes maps each degree to its allowed content types.
// A nil slice means all content types are allowed.
var DegreeContentTypes = map[string][]string{
	GameDegreeBeginner:     nil,
	GameDegreeIntermediate: {"block", "phrase", "sentence"},
	GameDegreeAdvanced:     {"sentence"},
}
